### PR TITLE
feat: Add excel export

### PR DESCRIPTION
**Pull request by igorsinc**
### SUMMARY
Add options to export charts data in `xlsx` format (similarly to already implemented `csv` export) with `xlsxwriter` library. `xls` export wasn't included in this PR as a legacy format.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/102555782/215083643-f1386f77-76ff-44e1-972f-4da19553affc.mov

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by grebennikov-undrew** [comment]
Hello, I have some questions:
1. Is it working on Dashboard? (not only in edit mode)
image

2. In what format are pivot tables exported? (raw or pivoted data)

Thank you!

**Comment by EugeneTorap** [comment]
@dpgaspar @villebro Can you review and merge the PR?

File: superset/common/query_context_processor.py [hunk 1]
```diff
@@ -446,15 +446,20 @@ def processing_time_offsets(  # pylint: disable=too-many-locals,too-many-stateme
...
-                df, index=include_index, **config["CSV_EXPORT"]
-            )
+
+            result = None
+            if self._query_context.result_format == ChartDataResultFormat.CSV:
+                result = csv.df_to_escaped_csv(
+                    df, index=include_index, **config["CSV_EXPORT"]
+                )
+            elif self._query_context.result_format == ChartDataResultFormat.XLSX:
+                result = excel.df_to_excel(df, **config["EXCEL_EXPORT"])
```
Comment (dpgaspar) [inline superset/common/query_context_processor.py:462]: on `df.to_excel(writer, **kwargs)` is encoding one of the kwargs?

**Review by dpgaspar — APPROVED** [review]

File: superset/common/query_context_processor.py (hunk 1 above)
Comment (EugeneTorap) [reply superset/common/query_context_processor.py:462]: At the moment yes, we do the same like for CSV!

Comment (EugeneTorap) [reply superset/common/query_context_processor.py:462]: @dpgaspar Can we merge the PR?

Comment (rusackas) [reply superset/common/query_context_processor.py:462]: Seems reasonable from where I sit... merging to unblock.

**Comment by mathieudruart** [comment]
Hi @EugeneTorap this is a nice feature, as mentioned here : https://github.com/apache/superset/issues/14415, is it planned to bring excel export to the dashboard?

**Comment by EugeneTorap** [comment]
Hi @mathieudruart! Thanks!
Feel free to open a PR for it.

**Comment by squalou** [comment]
Hi, 
discovered htat on 2.1.0-rc2 it's great.
I would have a question though : how would it be possible to choose a locale to export xls to ?
Meaning decimal separator mainly.

Maybe it's something feasible globally in superset, I didn't find.

**Comment by squalou** [comment]
may I suggest this

https://github.com/apache/superset/issues/23375

**Comment by Habeeb556** [comment]
Hello, is it possible to incorporate Excel functionality into SQLab as well? #25863

**Comment by rusackas** [comment]
A PR for that would be awesome/welcomed!

**Comment by Habeeb556** [comment]
@igorsinc I got the row limit set for the chart was reached. #29409

**Pull request merged**